controllers/storagesystem_controller.go

```
@@ -187,6 +189,16 @@ func (r *StorageSystemReconciler) validateStorageSystemSpec(instance *odfv1alpha
 	return nil
 }
 
+func (r *StorageSystemReconciler) addReferenceToRelatedObjects(instance *odfv1alpha1.StorageSystem, object runtime.Object) {
+
+	objectRef, err := reference.GetReference(r.Scheme, object)
+	if err != nil {
+		return
+	}
+
+	_ = objectreferencesv1.SetObjectReference(&instance.Status.RelatedObjects, *objectRef)
```

Similarly here, *do not ignore errors*.

---

Done